Loop load Jewish Calendar platforms

yuvalabou opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
loop load platforms, also removed unused code

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

yuvalabou committed:
Loop load Jewish Calendar platforms

tsvi commented:
LGTM

thecode commented on homeassistant/components/jewish_calendar/binary_sensor.py:
```diff
@@ -34,9 +34,6 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Jewish Calendar binary sensor devices."""
-    if discovery_info is None:
```
This is needed to protect against incorrect configuration such as:
```yaml
binary_sensor:
  - platform: jewish_calendar
 ```

Removing it may cause an exception for the above configuration:
```
2022-04-27 22:29:13 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up jewish_calendar platform for binary_sensor
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/workspaces/core/homeassistant/components/jewish_calendar/binary_sensor.py", line 40, in async_setup_platform
    async_add_entities([JewishCalendarBinarySensor(hass.data[DOMAIN], BINARY_SENSORS)])
KeyError: 'jewish_calendar'
```

thecode commented on homeassistant/components/jewish_calendar/sensor.py:
```diff
@@ -138,9 +138,6 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Jewish calendar sensor platform."""
-    if discovery_info is None:
```
Same as above

thecode reviewed (changes requested):
Left some comments about discovery info.
Also while testing this PR I have found out that it is missing a check if domain is missing in config, please add the following to `async_setup`:
```python
    if DOMAIN not in config:
        return True
```
An example for this: 
https://github.com/home-assistant/core/blob/c8c27dba0f569b23b94fc3e8b7d8b62898866ee6/homeassistant/components/modbus/__init__.py#L341-L344

yuvalabou committed:
Address review

yuvalabou commented:
Thanks Shay,
Reverted the cleanup and added your suggestion, although this case might be very rare.
Also sneaked in minor change in the sensor semantics as these are "info" rather than "data"

thecode commented:
If you want to increase coverage for discovery info you can add this test to `test_binary_sensor.py` and a similar one to `test_sensor.py`

```python
from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN

async def test_no_discovery_info(hass, caplog):
    """Test setup without discovery info."""
    assert BINARY_SENSOR_DOMAIN not in hass.config.components
    assert await async_setup_component(
        hass,
        BINARY_SENSOR_DOMAIN,
        {BINARY_SENSOR_DOMAIN: {"platform": jewish_calendar.DOMAIN}},
    )
    await hass.async_block_till_done()
    assert BINARY_SENSOR_DOMAIN in hass.config.components
```

This test simulates setting the platform using:
```yaml
binary_sensor:
  - platform: jewish_calendar
 ```

And checks that no error generated during setup.

thecode reviewed (approved):
Thanks!
